Dockerfile

```diff
@@ -1,19 +1,26 @@
-FROM node:14 as builder
+FROM node:14-alpine as builder
 WORKDIR /app
 COPY package*.json ./
-RUN npm install
+RUN npm ci
 COPY . .
-ARG APP_MOUNT_URI
+
 ARG API_URI
+ARG APP_MOUNT_URI
 ARG MARKETPLACE_URL
 ARG SALEOR_APPS_ENDPOINT
 ARG STATIC_URL
+
 ENV API_URI ${API_URI:-http://localhost:8000/graphql/}
 ENV APP_MOUNT_URI ${APP_MOUNT_URI:-/dashboard/}
+ENV MARKETPLACE_URL ${MARKETPLACE_URL}
+ENV SALEOR_APPS_ENDPOINT=${SALEOR_APPS_ENDPOINT}
 ENV STATIC_URL ${STATIC_URL:-/dashboard/}
-RUN STATIC_URL=${STATIC_URL} API_URI=${API_URI} MARKETPLACE_URL=${MARKETPLACE_URL} SALEOR_APPS_ENDPOINT=${SALEOR_APPS_ENDPOINT} APP_MOUNT_URI=${APP_MOUNT_URI} npm run build
 
-FROM nginx:stable
+RUN npm run build
+
+FROM nginx:stable-alpine as runner
 WORKDIR /app
+
 COPY ./nginx/default.conf /etc/nginx/conf.d/default.conf
+COPY ./nginx/replace-api-url.sh /docker-entrypoint.d/replace-api-url.sh
```

Comment:
Please follow the convention, filenames in `/docker-entrypoint.d` should start with a two digit number. Ex: `01-something.sh`

As suggested in https://github.com/tofran/saleor-dashboard/pull/3, please do:
`COPY ./nginx/replace-api-url.sh /docker-entrypoint.d/50-replace-api-url.sh`

I choose 50 since it runs after the built-in scripts and is flexible for expandability (end-users of your image).

Reply:
👋🏻 Hi @tofran and thank you for your comment. Can you send me a link to a convention about 2-digit numbers? So I will know for the future 😁

Reply:
First, you can see that in the `/docker-entrypoint.d` directory they already have a few scripts that follow this pattern:

```
$ docker run -it nginx:stable-alpine sh
/ # ls docker-entrypoint.d/
10-listen-on-ipv6-by-default.sh 
30-tune-worker-processes.sh
20-envsubst-on-templates.sh
```

Then if you have a look into their `docker-entrypoint.sh`, you will see that they intentionally sort scripts: 
```
find "/docker-entrypoint.d/" -follow -type f -print | sort -V | while read -r f; do
```

This is a way to control the execution flow if entry-point scrips. Alone should be enough, we should respect the upstream, but in Linux this scripting ordering is quite common.